pkg/eventsourcing/event_data.go

```diff
@@ -52,10 +52,20 @@ func (d EventData) toAny() (*anypb.Any, error) {
 }
 
 // ToProto unmarshals the contents the EventData into m.
-func (d EventData) ToProto(m protoreflect.ProtoMessage) error {
+func (d EventData) ToProto(m proto.Message) error {
 	a, err := d.toAny()
 	if err != nil {
 		return err
 	}
 	return a.UnmarshalTo(m)
 }
+
+// Unmarshal deserializes the EventData into a proto message using a type resolved from
+// the type URL.
+func (d EventData) Unmarshal() (proto.Message, error) {
```

Comment:
What is the difference between this method starting in line 65 and the method in line 55. Isn't that despite the style the same result? I think I do not understand the use-case for the additional method.

Reply:
The difference is that in the `ToProto`-method requires you to know the underlying proto message beforehand, whereas the `Unmarshal`-method resolves the type from the message itself. The usecase is that you have an event of a specific type, where the data itself is polymorphic, so that you have multiple possible underlying messages for the same event type.